### PR TITLE
fix(lsp): slice strings by byte index in code actions

### LINKED ISSUE #23361: Thread 'main' panicked at cli/lsp/analysis.rs:1132:27
Deno has panicked. This is a bug in Deno. Please report this
at https://github.com/denoland/deno/issues/new.
If you can reliably reproduce this panic, include the
reproduction steps and re-run with the RUST_BACKTRACE=1 env
var set and include the backtrace in your report.

Platform: linux x86_64
Version: 1.42.1
Args: ["/home/ec2-user/.deno/bin/deno", "lsp"]

thread 'main' panicked at cli/lsp/analysis.rs:1132:27:
byte index 5 is not a char boundary; it is inside '\u{a0}' (bytes 4..6) of `     clearInterval(this.timer);`
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[Error - 11:11:10 AM] The Deno Language Server server crashed 5 times in the last 3 minutes. The server will not be restarted. See the output for more information.

### PR DESCRIPTION
**Pull request by nayeemrmn**
Fixes #23361.

### CODE REVIEW

**Review by dsherret — APPROVED** [review]
LGTM

File: cli/lsp/analysis.rs [hunk 1]
```diff
@@ -1126,8 +1126,10 @@ impl CodeActionCollection {
 /// Prepend the whitespace characters found at the start of line_content to content.
 fn prepend_whitespace(content: String, line_content: Option<String>) -> String {
   if let Some(line) = line_content {
-    let whitespaces =
-      line.chars().position(|c| !c.is_whitespace()).unwrap_or(0);
+    let whitespaces = line
```
Comment (dsherret) [inline cli/lsp/analysis.rs:1129]: `whitespace_index`?

Comment (dsherret) [reply cli/lsp/analysis.rs:1129]: oh, I mean `non_whitespace_index`

Comment (nayeemrmn) [reply cli/lsp/analysis.rs:1129]: I guess it's like number of whitespace chars, probably fine as it is?

Comment (dsherret) [reply cli/lsp/analysis.rs:1129]: The name sounds like it's a vector or something plural.

**Pull request merged**